File: sol/solution/0001-0100/0038/Solution.swift
class Solution {
    func countAndSay(_ n: Int) -> String {
        var res = "1"
        if n <= 1 { return res }
        for _ in 2...n {
            var cur = ""
            var cnt = 1
            let chars = Array(res)
            for j in 1..<chars.count {
                if chars[j] == chars[j-1] {
                    cnt += 1
                } else {
                    cur += "\(cnt)\(chars[j-1])"
                    cnt = 1
                }
            }
            cur += "\(cnt)\(chars.last!)"
            res = cur
        }
        return res
    }
}

File: sol/solution/0001-0100/0038/Solution.go
func countAndSay(n int) string {
    res := "1"
    for i := 2; i <= n; i++ {
        cur := ""
        cnt := 1
        for j := 1; j < len(res); j++ {
            if res[j] == res[j-1] {
                cnt++
            } else {
                cur += strconv.Itoa(cnt) + string(res[j-1])
                cnt = 1
            }
        }
        cur += strconv.Itoa(cnt) + string(res[len(res)-1])
        res = cur
    }
    return res
}